axelor-human-resource/src/main/java/com/axelor/apps/hr/web/EmployeeController.java

```diff
@@ -149,4 +153,22 @@ public void updateEmployeeFilesWithMedicalVisit(ActionRequest request, ActionRes
     response.setValue(
         "employeeFileList", Beans.get(MedicalVisitService.class).addToEmployeeFiles(employee));
   }
+
+  @ErrorException
+  public void setDomainAnalyticDistributionTemplate(ActionRequest request, ActionResponse response)
+      throws AxelorException {
+    Context context = request.getContext();
+    Employee employee = context.asType(Employee.class);
+
+    EmploymentContract mainContract = employee.getMainEmploymentContract();
+
+    if (mainContract != null) {
```

Comment:
Here can you create a variable of company with Optional.of(employee).map(MainEmploymentContract).map(PayCompany).orElse(null)
and use the attr anytime using this company